v1/protocol_binding/binder.h

```diff
@@ -0,0 +1,157 @@
+#ifndef CLOUDEVENTSCPPSDK_V1_BINDING_BINDER_H
+#define CLOUDEVENTSCPPSDK_V1_BINDING_BINDER_H
+
+#include <memory>
+#include <regex> 
+
+#include <google/protobuf/message.h>
+
+#include "third_party/statusor/statusor.h"
+#include "third_party/base64/base64.h"
+#include "proto/cloud_event.pb.h"
+#include "external/googleapis/google/pubsub/v1/pubsub.pb.h"
+#include "v1/event_format/structured_cloud_event.h"
+#include "v1/event_format/json_formatter.h"
+#include "v1/util/formatter_util.h"
+#include "v1/util/cloud_events_util.h"
+
+namespace cloudevents {
+namespace binding {
+
+/* 
+ * Template for marshalling between protocol-specific Messages and CloudEvents.
+ * Each supported ProtocolBinding will implement template specialization 
+ * for the protected functions.
+ *
+ * User will interact with the public method functions Bind() and Unbind().
+ * Bind() will marhsal a CloudEvent to a protocol-specific Message.
+ * Unbind() will marshal a protocol-specific message to a CloudEvent.
+ * To create a StructuredContentMode Message, pass a EventFormat parameter to Bind().
+ * Code samples available in README.md.
+ */
+template <class Message>
+class Binder {
+    public:
+        // Create BinaryContentMode Message containing CloudEvent
+        absl::StatusOr<Message> Bind(io::cloudevents::v1::CloudEvent cloud_event) {
+            return BindBinary(cloud_event);
+        }
+
+        // Create StructuredContentMode Message containing Format-serialized CloudEvent  
+        absl::StatusOr<Message> Bind(io::cloudevents::v1::CloudEvent cloud_event, cloudevents::format::Format format) {
+            absl::StatusOr<std::unique_ptr<cloudevents::format::Formatter>> get_formatter;
+            get_formatter = cloudevents::formatter_util::FormatterUtil::GetFormatter(format);
+            if (!get_formatter.ok()) {
+                return get_formatter.status();
+            }
+            
+            absl::StatusOr<cloudevents::format::StructuredCloudEvent> serialization;
+            serialization = (*get_formatter) -> Serialize(cloud_event);
```

Comment:
remove spaces please.

Reply:
Which spaces? The indentation in front of the code line?